ray-operator/controllers/raycluster_controller.go

```diff
@@ -578,7 +578,7 @@ func (r *RayClusterReconciler) reconcileAutoscalerServiceAccount(instance *rayio
 	}
 
 	serviceAccount := &corev1.ServiceAccount{}
-	namespacedName := types.NamespacedName{Namespace: instance.Namespace, Name: instance.Name}
+	namespacedName := types.NamespacedName{Namespace: instance.Namespace, Name: instance.Spec.HeadGroupSpec.Template.Spec.ServiceAccountName}
```

Comment:
1. here, if the sa exists, it skip creating the new one.
2. you need to make sure head pod uses this service account name. -> I feel this part is missing?
@davidxia

Reply:
Thanks. I think I fixed 1 [here](https://github.com/ray-project/kuberay/pull/259/files#diff-37fdcfea947826e2a73f5ddcc0691ede6b513d411d6be53c3e529b950205e981R24) and 2 [here](https://github.com/ray-project/kuberay/pull/259/files#diff-8126a5e942d7e471be0f6a8e07cd26a9f4b9380b80c63ffc623f2ce33725a524R47).